### ISSUE TITLE
Problem in reproducing results on Avenue

### ISSUE THREAD

**Issue by NeeluMadanCreate** [LiUzHiAn/hf2vad#3]
Hi,

Thank you for sharing your contribution to anomaly detection. I tried to reproduce the baseline on the Avenue dataset. However, I got 89.06% AUC after retraining on Avenue. I got only 90.3% even with the pre-trained model. In the paper, it is 91.1%. Could you please share what could be a possible cause for this mismatch?

Thanks!
Neelu

**Comment by LiUzHiAn** [comment]
Hi,
 
I run the evaluation code again, as ` python eval.py --model_save_path=./pretrained_ckpts/avenue_HF2VAD_91.15.pth  --cfg_file=./pretrained_ckpts/avenue_HF2VAD_91.15_cfg.yaml`, the result is still 0.91147.

Did you following the same preprocessing procedure? And did you download the right pre-trained weights and the corresponding config file?

**Comment by LiUzHiAn** [comment]
![auc](https://user-images.githubusercontent.com/32020216/135376194-09fd4764-c28c-4545-b614-0977b70a4357.png)

**Comment by NeeluMadanCreate** [comment]
Thanks for your reply,
I followed the same pre-processing steps you mentioned on the repository; however, I was wondering how did you get "Test001_gt"? I didn't get them anyhow using the same preprocessing steps. I thought GT file is already inside "ground_truth_demo" folder. Could you please check?

![image](https://user-images.githubusercontent.com/80719339/135604454-342aaf83-1797-4eb9-9ea7-82cbd0b60504.png)

Thanks again!

**Comment by LiUzHiAn** [comment]
These pixel-level ground truth masks were officially provided by the UCSD Ped2 dataset, you can download them from the dataset page.

The provided `gt_label.json` in this repo was exactly built by these pixel-level masks. So, you don't need to get these `XXX_gt` folders actually.

**Comment by LiUzHiAn** [comment]
@NeeluMadanCreate,

1. You are right and thanks for pointing out this. I've updated the `finetune_cfg.yaml` file.
2. Not exactly, for ShanghaiTech, it should be `mem_usage: [ False, False, True, True ], skip_ops: [ "none", "none", "concat" ]`.

You can refer to the config files [here](https://drive.google.com/drive/folders/10B7WmZmBSgOPjkbedK9JwH6HRo06VSC2), and  use the corresponding pre-trained weights to get 91.1% result.

**Comment by Meenn** [comment]
Hi,Thank you for your contribution and for providing the code!
I followed the same pre-processing steps you mentioned on the repository to prepare the data.And i run eval with the pre-trained weights for ped2 and avenue.As a result,i got the 0.993078 for ped2 and 0.903187 for avenue.
I got the right result for peds.I think it means the process for preparing the data is right?Could you please share what could be a possible cause for the result of avenue(0.903187)?

**Comment by LiUzHiAn** [comment]
Did you use the corresponding config file (say, `avenue_HF2VAD_91.15_cfg.yaml`) for avenue?

**Comment by Meenn** [comment]
Thanks for your reply.
Yes,i have download the the right pre-trained weights and the corresponding config file,and i run eavl as 
"python eval.py \
         --model_save_path=./pretrained_ckpts/avenue_HF2VAD_91.15.pth \
         --cfg_file=./pretrained_ckpts/avenue_HF2VAD_91.15_cfg.yaml"
Finally i got 0.903187 but not 0.91147 as you mentioned.Could you please check if you can get the result of 0.91147 using pre-trained weights from [here](https://drive.google.com/drive/folders/10B7WmZmBSgOPjkbedK9JwH6HRo06VSC2)?

Thanks!

**Comment by LiUzHiAn** [comment]
Could you please leave your email, which will be more convenient to share some files.

**Comment by Meenn** [comment]
Of course,my email is mengxiaoxi@csu.edu.cn.
I did download the ckpt and config file from google cloud,but i can't get the expected results.Could i thought the whole process is right because of the correct result of ped2.
However,i don't know why i can't get the expected result of avenue.Could you please share what could be any other possible cause?
Thank you very much!

**Comment by LiUzHiAn** [comment]
Hi, guys

Thanks for pointing this issue. It was caused by a little bug during preprocessing. For the testing samples, I saved them into a single chunk file and extract 100K samples by default (check [here](https://github.com/LiUzHiAn/hf2vad/blob/f909f427001e7bbabf8994e8992bd3e72334f547/pre_process/extract_samples.py#L11) for details). But this would miss some samples if the total sample number is greater than 100K, hence leading to the inconsistent AUROC results.

I've fixed this issue [here](https://github.com/LiUzHiAn/hf2vad/blob/master/pre_process/extract_samples.py#L11-#L18) and run the evaluation phase again, and the results can be obtained.

Thank you.
